openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -533,3 +535,33 @@ def _validate_email(email):
         raise AccountEmailInvalid(
             u"Email '{email}' format is not valid".format(email=email)
         )
+
+
+@intercept_errors(UserAPIInternalError, ignore_errors=[UserAPIRequestError])
+@transaction.atomic
+def delete_user(user):
+    """
+    Delete a user and any related records.
+
+    Related records include the following:
+    1. models with a ForeignKey relationship to User (with on_delete=CASCADE)
+    2. models with indirect relationship to User (i.e. through another model, or with on_delete other than CASCADE)
+    3. files belonging to the user
+
+    Arguments:
+        user: The User object
+
+    Returns:
+        None
+
+    Raises:
+        UserAPIInternalError
+    """
+    # delete profile images
+    delete_profile_images.delay([user.username])
```

Comment:
@john2x, wouldn't it be better to return the task ID and provide an API to check the status of the task by ID?

Reply:
@lgp171188 Is there an existing api to check the status of Celery tasks? I could hook into that. Also there might be additional tasks being queued in the future (e.g. certificates, assignments), so the return value would need to be a dict or similar.